index.html

```diff
@@ -0,0 +1,464 @@
+          [=list=]:
+            <ol class="algorithm">
+              <li>[=list/For each=] |entry| of |json_launch_handler|["route_to"]:
+                <ol class="algorithm">
+                  <li>If the type of |entry| is not [=string=], continue.
+                  </li>
+                  <li>If |json_launch_handler|["route_to"] is supported by the user agent, set manifest_launch_handler["route_to"] to |entry| and return.
+                  </li>
+                </ol>
+              </li>
+            </ol>
+          </li>
+          <li>If the type of |json_launch_handler|["route_to"] is not [=string=], return.
+          </li>
+          <li>If |json_launch_handler|["route_to"] is supported by the user agent, set manifest_launch_handler["route_to"] to |json_launch_handler|["route_to"].
+          </li>
+        </ol>
+        <p class="note">
+          `route_to` accepts a list of strings as a future proofing mechanism. If a new [=route to target=] is added in the future but has not yet been implemented by all user agents, web developers can cater to the old user agents by providing fallback [=route to target=] values in a list format. User agents will use the first value that is valid to them.
+        </p>
+        <p class="note">
+          For example: if "some-new-behaviour" is added in the future a site developer can specify "[=route to/existing-client-navigate=]" to be the fallback with the following:
+        </p>
+        <pre class="note json">
+          {
+            "launch_handler": {
+              "route_to": ["some-new-behaviour", "existing-client-navigate"]
+            }
+          }
+        </pre>
+      </section>
+    </section>
+    <section>
+      <h2>
+        Handling Web App Launches
+      </h2>
+      <section>
+        <h2>
+          Launching a Web App
+        </h2>
+        <p>
+          When launching a web app via any means (e.g. OS launch, protocol handling, file handling, share target, link capturing, etc.) the trigger of the launch should run the steps to [=launch a web app=] with a {{LaunchParams}} containing relevant launch information (e.g. share data in the event of a share target launch).
+        </p>
+        <p>
+          The steps to <dfn>launch a web app</dfn> are given by the following algorithm and takes a {{LaunchParams}} |params:LaunchParams|.
+        </p>
+        <ol class="algorithm">
+          <li>If |params| is null, set |params| to a new {{LaunchParams}} with {{LaunchParams/targetURL}} set to [=manifest/start_url=].
+          </li>
+          <li>Set |client| to the result of running the steps to [=prepare a web app launch client=] passing [=manifest/route_to=] and |params|.{{LaunchParams/targetURL}}.
+          </li>
+          <li>Append |params| to the [=unconsumed launch params=] of the |client|'s document's {{Window.LaunchQueue}}.
+          </li>
+          <li>If the [=assigned launch consumer=] |consumer| is set:
```

Comment:
Launch navigation optionally happens in `prepare a web app launch client`. If there's no `LaunchParams` consumer then it's just added to the queue.

Reply:
So if a site has launch handlers in their manifest and doesn't register a consumer, launches do nothing?

Maybe we can call this bad case out somehow? Maybe in the explainer? I wonder if we can avoid this somehow.

Reply:
This is really only an issue for `existing-client-retain`, and yes it is by design. By using that route mode, you are explicitly opting in to having the default launch behaviour be nothing happens, and you have to handle it. We give you the tools to do that by ensuring you can hook up a launch consumer without a race condition.

We should explicitly call this out in the spec, and explainer, and anywhere else. There be dragons in `existing-client-retain`. (Also note that this is the reason why `auto` is not allowed to default to `existing-client-retain`, since the other two modes basically are reasonable user interface changes on top of HTML, whereas ECR breaks site functionality unless you participate. I would add a mention of this to the note under the definition of "auto".)

Reply:
Good idea, added a warning section to existing-client-retain.

Reply:
The new language LGTM in the warning.